src/TBuffer.cpp

```diff
@@ -1314,392 +619,29 @@ void TBuffer::translateToPlainText(std::string& incoming, const bool isFromServe
 
         // We are outside of a CSI or OSC sequence if we get to here:
 
-        if (mMXP && mpHost->mServerMXPenabled && (mMXP_MODE != MXP_MODE_LOCKED)) {
-
-            // ignore < and > inside of parameter strings
-            if (openT == 1) {
-                if (ch == '\'' || ch == '\"') {
-                    if (!mParsingVar) {
-                        mOpenMainQuote = ch;
-                        mParsingVar = true;
-                    } else {
-                        if (ch == mOpenMainQuote) {
-                            mParsingVar = false;
-                        }
-                    }
-                }
-            }
-
-            if (ch == '<') {
-                if (!mParsingVar) {
-                    ++openT;
-                    if (!currentToken.empty()) {
-                        currentToken += ch;
-                    }
-                    mAssemblingToken = true;
-                    ++localBufferPosition;
+        if (mpHost->mMxpProcessor.isEnabled() && mpHost->mServerMXPenabled) {
+            if (mpHost->mMxpProcessor.getMode() != MXP_MODE_LOCKED) {
+                TMxpProcessingResult result = mpHost->mMxpProcessor.processMxpInput(ch);
+                if (result == HANDLER_NEXT_CHAR) {
+                    localBufferPosition++;
                     continue;
+                } else if (result == HANDLER_COMMIT_LINE) { // BR tag
+                    ch = '\n';
+                    goto COMMIT_LINE;
+                } else { //HANDLER_FALL_THROUGH -> do nothing
+                    assert(localBuffer[localBufferPosition] == ch);
+                    // the curent char may have changed if it is the last char of an entity such as &gt; ?
+//                    localBuffer[localBufferPosition] = ch;
```

Comment:
Previously the code in this loop would process a char and could do one of the following:
1. continue; -> read the next char in the loop
2. return; -> returns frin the function (skip the remaining of the current buffer)
3. follow processing to the next step
4. jump to the commit line phase (notably when BR tag is identified)

The rest of the code in this function looks similar (but usually just continue; or fallthrough). 

Before my refactoring this was a huge chunk that checked for these conditions and directly executed (continue; return; goto COMMIT_LINE; (nothing)). Since I moved the code somewhere else I needed to have some way to still get the same result. So I return a enum value (NEXT_CHAR, COMMIT_LINE or FALLTHROUGH) to represent the action that should be performed in the loop.

The "return;" case was buggy.